### DISCUSSION TITLE
handler that responds to get and head

### DISCUSSION THREAD

**Discussion by mlilback** [actix/actix-web#2608] [Q&A] [answered]
I"m currently using the following pattern for my handlers:
```
#[get("/{book_id}/fsitems")]
pub async fn files(path: web::Path<i64>, db_pool: web::Data<Pool>) -> Result<HttpResponse, NCError> {
...
}

pub fn routes() -> Scope {
    web::scope("/book")
        .service(handlers::files)
}
```

How do I make this handler also respond to head requests?

All the examples I could find use `scope.guard(...).to(|| HttpResponse::Ok())`, which won't take my handler for the `to` argument.

**Comment by robjtede** [comment, marked as answer]
https://docs.rs/actix-web/latest/actix_web/attr.route.html

**Reply by mlilback** [reply]
Much appreciated

**Reply by D0bhareach** [reply]
Not sure if it's a correct approach. 
```rust
async fn root_handler() -> HttpResponse {
    HttpResponse::Ok()
        .content_type(ContentType::plaintext())
        .body("test1: index")
}

App::new().wrap(cors).service(
            scope("/test")
                .service(resource("").get(root_handler).head(root_handler))
                .route("/hello", web::get().to(hello)),
        )
``` 
while using #[route] caused: `root_handler: actix_web::Handler<_>` is not satisfied the trait `actix_web::Handler<_>` is not implemented for `root_handler`

**Reply by robjtede** [reply]
When using `#[route]` you'd need to register the handler like this: `.service(root_handler)`.

**Reply by D0bhareach** [reply]
Thank you. Works well. For documentation here are lines that changed:
```rust
// bring macro to scope
use actix_web::route;

#[route("", method = "GET", method = "HEAD")] //  this is new
async fn root_handler() -> HttpResponse {todo!()} // same

scope("/test-1") // same
    .service(root_handler) // this as suggested
```
